docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,202 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+From a developer's perspective, the C#/VB async model is some nice syntax that hides complexity and allows you to focus on writing methods without worrying about things like thread pools and callbacks.
```

Comment:
You are missing the value-prop. While it's true that it is nice syntax that hides complexity, it doesn't state what the end goal is. What does it all achieve?